awesome/src/objects/tag.rs

```diff
@@ -326,4 +420,123 @@ assert(called_selected == 1)
                  None
         ).unwrap()
     }
+
+    #[test]
+    fn tag_client() {
+        let lua = Lua::new();
+        tag::init(&lua).unwrap();
+        lua.eval(
+             r#"
+local t = tag{}
+assert(#t:clients() == 0, "Cannot get the clients")
+"#,
+             None
+        ).unwrap()
+    }
+
+    #[test]
+    fn tag_set_client() {
+        let lua = Lua::new();
+        tag::init(&lua).unwrap();
+        client::init(&lua).unwrap();
+        lua.eval(
+             r#"
+local c = client{}
+local t = tag{}
+t:clients({ c })
+assert(c, "client doesn't exists")
+assert(#t:clients() == 1, "Tag doesn't have the clients")
+assert(t:clients()[1] == c, "Pass by value, not by reference")
```

Comment:
You should be modifying `c` in between this and the insert call shouldn't you to assure it is actually by reference?